lib/BasicTypes.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * BasicTypes.cpp, part of VCMI engine
```

Comment:
Filename basicTypes a bit misleading. Maybe do you have plans to extend it with more functionality? can you share you plans?

Reply:
I want to extract more basic types when I will do HeroBonus.cpp split. 
1. something like ICommonFactionMember (which will have damage, attack, defence and so on)
2. move some common methods from C* types to Entity types (if this methods can be implemented purely only via entity interface, not using any additional state). Of course, methods will remain to be virtual.